DataDog/datadog-agent
# Simplify the Forwarder interface by removing the argument `priority`

### What does this PR do?

Simplify the Forwarder interface by removing the argument `priority`

Note: `SubmitMetadata` and `SubmitHostMetadata` were unused.
Before, `Serializer.SendMetadata` and `Serializer.SendHostMetadata` called `Forwarder.SubmitV1Intake`.
Now  these 2 functions call respectively `Forwarder.SubmitMetadata` and `Forwarder.SubmitHostMetadata` which perform the same as `Forwarder.SubmitV1Intake` (`Forwarder.SubmitHostMetadata` use TransactionPriorityHigh)

### Describe your test plan
The code should behave the same as before.

- Check metadata and hostmetadata payloads are send correctly.
- Check the hostmetadata payloads are always kept in the forwarder retry queue (because of TransactionPriorityHigh)
